config.go

```diff
@@ -174,9 +174,22 @@ func getPluginVersion() string {
 	return "unknown"
 }
 
-func isValidIAM(cident *string) bool {
+func isValidIAM(arn *string, client *alks.Client) bool {
 
-	if strings.Contains(*cident, "assumed-role/Admin/") || strings.Contains(*cident, "assumed-role/IAMAdmin/") {
+	/*
+		Validates ARN for assumed-role of:
+			- Admin
+			- IAMAdmin
+			- Machine Identities.
+	*/
+	if strings.Contains(*arn, "assumed-role/Admin/") || strings.Contains(*arn, "assumed-role/IAMAdmin/") {
+		return true
+	} else if strings.Contains(*arn, "arn:aws:sts::") {
```

Comment:
The get caller identity response will always contain `arn:aws:sts::` so we can probably drop this conditional.

Reply:
Here's an example of a non-human ARN response:
```sh
[ec2-user@ip-172-22-1-120 ~]$ aws sts get-caller-identity
{
    "Account": "805619180788",
    "UserId": "AROA3XEVFET2K6VSUL7IR:i-0fec6c0b834acdaa6",
    "Arn": "arn:aws:sts::805619180788:assumed-role/TempBastionRole/i-0fec6c0b834acdaa6"
}
```
I don't see any way to tell them apart, just sharing for context.

Reply:
Ah okay I completely forgot. I kept thinking it was just returning some generic ARN. :face-palm: